site/src/components/common/Button.tsx

```diff
@@ -0,0 +1,120 @@
+import { CallToActionBlockData } from "@src/blocks.generated";
+import { forwardRef, HTMLAttributes, RefObject } from "react";
+import styled, { css } from "styled-components";
+
+type ButtonVariant = CallToActionBlockData["variant"];
+
+type ButtonProps = {
+    variant?: ButtonVariant;
+    disabled?: boolean;
+} & (HTMLAttributes<HTMLButtonElement> | (HTMLAttributes<HTMLAnchorElement> & Pick<HTMLAnchorElement, "href" | "target">));
+
+export const Button = forwardRef<HTMLButtonElement | HTMLAnchorElement, ButtonProps>(
+    ({ variant = "Contained", disabled = false, children, ...htmlAttributes }, ref) => {
+        return "href" in htmlAttributes ? (
+            <StyledAnchor ref={ref as RefObject<HTMLAnchorElement>} $variant={variant} $disabled={disabled} {...htmlAttributes}>
+                {children}
+            </StyledAnchor>
+        ) : (
+            <StyledButton ref={ref as RefObject<HTMLButtonElement>} $variant={variant} disabled={disabled} {...htmlAttributes}>
+                {children}
+            </StyledButton>
+        );
+    },
+);
+
+const buttonVariantStyle: Record<ButtonVariant, ReturnType<typeof css>> = {
+    Contained: css`
+        ${({ theme }) => css`
+            background-color: ${theme.palette.primary.main};
+            color: ${theme.palette.primary.contrastText};
+            border: 1px solid ${theme.palette.primary.main};
+
+            &:hover {
+                background-color: ${theme.palette.primary.dark};
+                border: 1px solid ${theme.palette.primary.dark};
```

Comment:
You could just change the `border-color` here, as the rest does not change. 
E.g. if the width ever changes in a project, the value only has to be changed in one place then. 

```suggestion
                border-color: ${theme.palette.primary.dark};
```